src/models/xyhw.rs

```diff
@@ -268,6 +269,38 @@ impl Xyhw {
         .into()
     }
 
+    pub fn scratchpad_size_position(&self, scratch_pad: &ScratchPad) -> Xyhw {
+        let x_sane = if (10..90).contains(&scratch_pad.x) {
+            scratch_pad.x
+        } else {
+            25
+        };
+        let y_sane = if (10..90).contains(&scratch_pad.y) {
+            scratch_pad.y
+        } else {
+            25
+        };
+        let height_sane = if (10..90).contains(&scratch_pad.height) {
+            scratch_pad.height
+        } else {
+            50
+        };
+        let width_sane = if (10..90).contains(&scratch_pad.width) {
+            scratch_pad.width
+        } else {
+            50
+        };
+
+        XyhwBuilder {
+            x: self.x + self.w * x_sane / 100,
+            y: self.y + self.h * y_sane / 100, //- (self.h / 4),
+            h: self.h * height_sane / 100,
+            w: self.w * width_sane / 100,
+            ..XyhwBuilder::default()
+        }
+        .into()
+    }
+
```

Comment:
Having something scratchpad specific in `Xyhw` seems a bit weird, Maybe it should be the other way around?
Scratchpad can know about Xyhw, but Xyhw probably shouldn't have to know anything about Scratchpad.

Reply:
Because I still don't have a whole picture of the project, would you mind let me know the correct place for this? I don't want to screw the structure of the project.

Reply:
I checked that currently only config have a scratchpad module. But it's only for configuration.

Reply:
> Because I still don't have a whole picture of the project

We're on the same boat :wink: 
Unfortunately, you'll have to wait for another contributor to advise you on that, I'm not yet comfortable enough with the code to give you a clear suggestion.